sechub-scan-product-sereco/src/main/java/com/mercedesbenz/sechub/domain/scan/product/sereco/WebScanProjectDataFalsePositiveStrategy.java

```diff
@@ -0,0 +1,124 @@
+
+import com.mercedesbenz.sechub.commons.model.ScanType;
+import com.mercedesbenz.sechub.domain.scan.project.FalsePositiveProjectData;
+import com.mercedesbenz.sechub.domain.scan.project.WebscanFalsePositiveProjectData;
+import com.mercedesbenz.sechub.sereco.metadata.SerecoVulnerability;
+import com.mercedesbenz.sechub.sereco.metadata.SerecoWeb;
+
+@Component
+public class WebScanProjectDataFalsePositiveStrategy implements SerecoProjectDataFalsePositiveStrategy {
+
+    private static final Logger LOG = LoggerFactory.getLogger(WebScanProjectDataFalsePositiveStrategy.class);
+
+    @Autowired
+    SerecoProjectDataWebScanFalsePositiveSupport webscanFalsePositiveProjectDataSupport;
+
+    @Override
+    public boolean isFalsePositive(SerecoVulnerability vulnerability, FalsePositiveProjectData projectData, Map<String, Pattern> projectDataPatternMap) {
+        notNull(vulnerability, " vulnerability may not be null");
+        notNull(projectData, " projectData may not be null");
+        notNull(projectDataPatternMap, " projectDataPatternMap may not be null");
+
+        // We use a fast exit approach in this method, since all conditions must be
+        // satisfied, we exit as soon as possible.
+
+        if (projectDataPatternMap.isEmpty()) {
+            return false;
+        }
+
+        if (vulnerability.getScanType() != ScanType.WEB_SCAN) {
+            return false;
+        }
+
+        WebscanFalsePositiveProjectData webScanData = projectData.getWebScan();
+        if (webScanData == null) {
+            return false;
+        }
+
+        SerecoWeb vulnerabilityWeb = vulnerability.getWeb();
+        if (vulnerabilityWeb == null) {
+            LOG.error("Cannot check web vulnerability for false positives when vulnerability data has no web parts!");
+            return false;
+        }
+
+        /* ---------------------------------------------------- */
+        /* -------------------CWE ID--------------------------- */
+        /* ---------------------------------------------------- */
+        if (!webscanFalsePositiveProjectDataSupport.areBothHavingSameCweIdOrBothNoCweId(webScanData, vulnerability)) {
+            return false;
+        }
+
+        String target = vulnerabilityWeb.getRequest().getTarget();
+        if (target == null) {
+            return false;
+        }
+
+        URL targetUrl = null;
+        try {
+            targetUrl = new URL(target.trim());
+        } catch (MalformedURLException e) {
+            LOG.error("Sereco vulnerability webscan target URL: {} is not a valid URL!", target, e);
+            return false;
+        }
+
+        /* ---------------------------------------------------- */
+        /* -------------------SERVERS-------------------------- */
+        /* ---------------------------------------------------- */
+        String host = targetUrl.getHost();
+        if (!webscanFalsePositiveProjectDataSupport.isMatchingHostPattern(host, webScanData.getHostPatterns(), projectDataPatternMap)) {
```

Comment:
This is intentional, since if it cannot be a false positive no further checks are necessary. In case it could be a false positive at this poit the following requirements must be checked as well.